### ISSUE TITLE
Codegen: Add `Identifiable` conformance

### ISSUE THREAD

**Issue by designatednerd** [apollographql/apollo-ios#710]
Great idea from [the thread seeking Swift codegen suggestions](https://github.com/apollographql/apollo-ios/issues/682#issuecomment-520856397) - anything that has a GraphQLID should conform to `Identifiable`. 

Note: `Identifiable` is part of the Swift 5.1 release, so it will need to wait until that gets finalized. That's likely well before the updated codegen gets finalized, though.

**Comment by rodrigoelp** [comment]
Ideally it should auto implement `Hashable` as well, allowing us to use the objects/structs directly in dictionaries and/or sets.
However, I understand some people might say certain fields should be consider in the hash and some people might vote for adding everything to the hash.

**Comment by designatednerd** [comment]
@rodrigoelp Hashable and Equatable conformance will be added by default in updated codegen - please see the RFC in #939 for details.

**Comment by rodrigoelp** [comment]
Ah, bad @designatednerd, I did read that issue (only the part of `Equatable`). Thanks for the reply and happy new year.

**Comment by daisyramos317** [comment]
@designatednerd is there something missing in this query or a doc for GraphQLID I can take a look at? This project is running on Swift 5.6.1 but not seeing `Identifiable` conformance generated

```GraphQL
query getLiveStreams($status: LiveStreamStatus) {
 	liveStreams(status: $status) {
 		id
 		channel {
 			id
 			name
 		}
 	}
 }

**Comment by designatednerd** [comment]
Hi Daisy, I no longer work at Apollo so information on timeline is best gotten from @AnthonyMDev and @calvincestari, but this was not implemented in Original Crispy Flavor™ codegen, it may be coming in updated codegen but I remember running into some issues around it. 

Guys, any thoughts?

**Comment by AnthonyMDev** [comment]
This is not being implemented in the 1.0 release with the new Codegen. There are a lot of good reasons why this is not being included. Primarily, GraphQL objects should be stored using the GraphQL `NormalizedCache`, not using `Codable`, so there isn't really a great reason for this.

**Comment by dfalling** [comment]
@AnthonyMDev,

Is there a reason to not implement Identifiable as suggested in this ticket? It would make iterating over collections in SwiftUI a bit less boilerplatey:

### Current

```swift
ForEach(users, id: \.self.id) { user in
  NavigationLink(destination: UserDetails(user: user)) {
    Text(user.name)
  }
  .tag(user.id)
}
```

### With Identifiable

```swift
ForEach(users) { user in
  NavigationLink(destination: UserDetails(user: user)) {
    Text(user.name)
  }
}
```
